### PR TITLE
fix(packaging): increased buffer size

### LINKED ISSUE #15839: bug(packaging): gulp public-api:update throws stdout max buffer exceeded
With the current build trying to run `gulp public-api:update`  will cause `stdout max buffer exceeded` error.
Currently the max buffer size is 200mb which is the default.
Changing it to 300mb fixes this error (credit @progdom)

### PR DESCRIPTION
**Pull request by Progdom**
**Please check if the PR fulfills these requirements**
- [x ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
maxBuffer is exceeded when trying to run public-api:update. 

**What is the new behavior?**
maxBuffer is increased so that build has enough memory to complete the task.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
Closes #15839

### CODE REVIEW

**Comment by Progdom** [comment]
https://travis-ci.org/angular/angular/builds/219874166 says it's failing, but the logs don't show an error. All commands are returning code 0. Anyone know why this is?

File: tools/gulp-tasks/build.js [hunk 1]
```diff
@@ -2,5 +2,5 @@
 module.exports = (gulp) => (done) => {
   const path = require('path');
   const childProcess = require('child_process');
-  childProcess.exec(path.join(__dirname, '../../build.sh'), done);
+  childProcess.exec(path.join(__dirname, '../../build.sh'), {maxBuffer: 300 * 1024}, done);
```
Comment (vicb) [inline tools/gulp-tasks/build.js]: could you please add a comment here explaining why.
Thanks !

**Comment by vicb** [comment]
Thanks for the PR, could you please address the inline comment before we merge this ?

**Comment by Progdom** [comment]
comment added.

**Review by vicb — APPROVED** [review]

**Comment by Progdom** [comment]
bump

**Pull request merged**